server/auth/index.js

```diff
@@ -22,16 +22,20 @@ router.use('/', email.routes());
 router.get('/redirect', auth(), async ctx => {
   const user = ctx.state.user;
 
-  // transfer access token cookie from root to subdomain
-  ctx.cookies.set('accessToken', undefined, {
-    httpOnly: true,
-    domain: stripSubdomain(ctx.request.hostname),
-  });
-
-  ctx.cookies.set('accessToken', user.getJwtToken(), {
-    httpOnly: false,
-    expires: addMonths(new Date(), 3),
-  });
+  // transfer access token cookie from root to subdomain,
+  // only if it exists already. Otherwise we would
+  // convert an api-token to an access-token.
+  if (ctx.cookies.get('accessToken') !== '') {
```

Comment:
```suggestion
  if (ctx.cookies.get('accessToken')) {
```

Should this be a less strict check, thinking if the cookie is somehow returned as `null` or `undefined` this would also allow the redirect to continue

Reply:
On that note: Do we want to allow the redirect to continue if the cookie has any arbitrary value? Someone could create a cookie `accessToken` with any non-null value and then be redirected. Does it make sense the compare the existing cookie value and the value we would be setting (`user.getJwtToken()`) and only proceed if they are equal?

Reply:
> Do we want to allow the redirect to continue if the cookie has any arbitrary value?

Hard to see any harm in this, the subdomains are "public" information so you wouldn't achieve anything by doing that.

> Does it make sense the compare the existing cookie value and the value we would be setting (user.getJwtToken()) and only proceed if they are equal?

I think this is the solid move 👍

Reply:
I went ahead and made this change, pushed as f4c4a112